plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/nessie/IcebergNessieCatalogConfig.java

```diff
@@ -168,4 +169,17 @@ public boolean isMissingTokenForBearerAuth()
     {
         return getSecurity().filter(isEqual(BEARER)).isEmpty() || getBearerToken().isPresent();
     }
+
+    public Optional<String> getClientAPIVersion()
+    {
+        return clientAPIVersion;
+    }
+
+    @Config("iceberg.nessie-catalog.client-api-version")
+    @ConfigDescription("Optional client api version to be used instead of inferring from server URI")
```

Comment:
that's too generic. there is no v3 or v2 so accepting a string here doesn't make sense. Let's call it `use-v2-api` instead (boolean)

Reply:
Can we have an enum instead ?

Reply:
done